cmd/metrics.go

```diff
@@ -64,24 +90,54 @@ var Metrics = metrics
 /*
 Produce data for given metrics.
 */
-func metrics(filePath string, metricsItems []string, readFile ReadFileFunc) []byte {
-	jsonData := readFile(filePath)
-	adjacencyList := loadJsonFile(jsonData)
+func metrics(filePathDg string, filePathDgReverse string, metricsItems []string, readFile ReadFileFunc) []byte {
+	var adjacencyList AdjacencyList
+	var adjacencyListReverse AdjacencyList
+
 	report := make(map[string]map[string]interface{})
+	// use dependencies adjacency list as is
+	if slices.Contains(metricsItems, MetricDependenciesDirect) || slices.Contains(metricsItems, MetricDependenciesTransitive) {
+		jsonData := readFile(filePathDg)
+		adjacencyList = loadJsonFile(jsonData)
+	}
+	// use the reversed dependencies list if provided otherwise reverse the dependencies list first
+	if slices.Contains(metricsItems, MetricReverseDependenciesDirect) || slices.Contains(metricsItems, MetricReverseDependenciesTransitive) {
+		if filePathDgReverse != "" {
+			jsonData := readFile(filePathDgReverse)
+			adjacencyListReverse = loadJsonFile(jsonData)
+		} else {
+			jsonData := readFile(filePathDg)
+			adjacencyListTemp := loadJsonFile(jsonData)
```

Comment:
Functions like `readFile` and `loadJsonFile` must return an error instead of panicking.